### PR TITLE
feat(models): added support for local models

### PR DESCRIPTION
**Pull request by cl-shaharshavit-rnd**

### CODE REVIEW

File: src/controllers/swagger.ts [hunk 1]
```diff
@@ -28,21 +30,72 @@ export class SwaggerController {
...
+    };
+
+    const logger = this.logger;
+    return resolveRefs(root, options).then(
+      function (results) {
+        return results.resolved as Record<string, unknown>;
+      },
+      function (err: Error) {
+        logger.warn(`failed to load swagger reference: ${err.message}.`);
+        console.log(err.stack);
```
Comment (asafMasa) [inline src/controllers/swagger.ts:97]: will we want this log in the logger or just in console?

File: docs/openapi3.yaml [hunk 2]
```diff
@@ -23,7 +23,7 @@ paths:
             application/json:
               schema:
                 allOf:
-                  - $ref: 'http://localhost/Schema/Http/ImageMetadataResponse.json'
+                  - $ref: '../Schema/Http/ImageMetadataResponse.json'
```
Comment (asafMasa) [inline docs/openapi3.yaml:26]: 1. do you think we need to rename to camel case?
2. why not ImageMetadata-Response as with the ImageMetadata-Create and -Update?

File: package.json [hunk 3]
```diff
@@ -38,6 +40,7 @@
     "@commitlint/cli": "^9.1.1",
     "@commitlint/config-conventional": "^10.0.0",
     "@map-colonies/eslint-config": "^1.1.0",
+    "@map-colonies/mc-model-types": "file:../mc-models/map-colonies-mc-model-types-1.0.2.tgz",
```
Comment (asafMasa) [inline package.json:43]: remove after publish

**Review by asafMasa — CHANGES_REQUESTED** [review]
👍 Great job

File: src/controllers/swagger.ts [hunk 4]
```diff
@@ -28,21 +30,72 @@ export class SwaggerController {
...
+      res.sendStatus(httpStatus.SERVICE_UNAVAILABLE);
+    }
   }
 
   public serveJson(req: Request, res: Response): void {
     res.json(this.swaggerDoc);
   }
 
+  private async init(): Promise<void> {
+    //this.swaggerDoc = load('./docs/openapi3.yaml') as swaggerUi.JsonObject;
```
[the remark this answers is not in the document]
Comment (cl-shaharshavit-rnd) [reply src/controllers/swagger.ts:52]: removed

File: src/controllers/swagger.ts [hunk 5]
```diff
@@ -28,21 +30,72 @@ export class SwaggerController {
...
+    const servers = doc.servers as SwaggerServer[];
     servers[0].url = `${host}:${port}`;
   }
+
+  private async multiFileSwagger(
+    root: Record<string, unknown>
+  ): Promise<void | Record<string, unknown>> {
+    const options = {
+      filter: ['relative', 'remote'],
+      location: './docs/openapi3.yaml', //'./Schema/root.json',
```
[the remark this answers is not in the document]
Comment (cl-shaharshavit-rnd) [reply src/controllers/swagger.ts:78]: removed

File: src/controllers/swagger.ts (hunk 1 above)
Comment (cl-shaharshavit-rnd) [reply src/controllers/swagger.ts:97]: removed

File: docs/openapi3.yaml (hunk 2 above)
Comment (cl-shaharshavit-rnd) [reply docs/openapi3.yaml:26]: 1) don't think it matters
2)that's the name of the file in models
2.1) the -create and -update in the models takes a base model and only set what fields are required. this is a composition of 2 different types

**Review by asafMasa — APPROVED** [review]

**Pull request merged**